wdl/transforms/draft3/src/test/scala/AstToWorkflowDefinitionElementSpec.scala

```diff
@@ -0,0 +1,97 @@
+import cats.data.Validated._
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+import wdl.model.draft3.elements.ExpressionElement._
+import wdl.model.draft3.elements.{InputDeclarationElement, InputsSectionElement, IntermediateValueDeclarationElement, OutputDeclarationElement, OutputsSectionElement, PrimitiveTypeElement}
+import wom.types.{WomSingleFileType, WomStringType}
+import wdl.transforms.base.ast2wdlom.AstToWorkflowDefinitionElement._
+
+
+class AstToWorkflowDefinitionElementSpec extends AnyFlatSpec{
+  behavior of "Check Stdouts and Stderrs"
+
+  val mockInputSectionStdout = Option(InputsSectionElement(Vector(InputDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "i", Some(StdoutElement)))))
+  val mockInputSectionStderr= Option(InputsSectionElement(Vector(InputDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "i", Some(StderrElement)))))
+  val mockInputSectionNonStd= Option(InputsSectionElement(Vector(InputDeclarationElement(PrimitiveTypeElement(WomStringType), "more", Some(StringLiteral("more"))))))
+
+
+  val mockIntermediatesStdout = Vector(IntermediateValueDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "y", StdoutElement))
+  val mockIntermediatesStderr = Vector(IntermediateValueDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "y", StderrElement))
+  val mockIntermediatesNonStd = Vector(IntermediateValueDeclarationElement(PrimitiveTypeElement(WomStringType), "here", StringLiteral("here")))
+
+  val mockOutputSectionStdout = Option(OutputsSectionElement(Vector(OutputDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "s", StdoutElement))))
+  val mockOutputSectionStderr = Option(OutputsSectionElement(Vector(OutputDeclarationElement(PrimitiveTypeElement(WomSingleFileType), "s", StderrElement))))
+  val mockOutputSectionNonStd = Option(OutputsSectionElement(Vector(OutputDeclarationElement(PrimitiveTypeElement(WomStringType), "more", StringLiteral("more")))))
+
+
+  it should "return an error when there is an stdout in input section" in {
+    val testInputs = checkIfStdInputExist(mockInputSectionStdout, StdoutElement, "stdout")
+    testInputs match {
+      case Valid(_) => fail("Input section contained stdout. Should have failed.")
+      case Invalid(e) => e.head should be("Workflow cannot have stdout expression in input section at workflow-level.")
+    }
+  }
+
+  it should "return an error when there is an stderr in input section" in {
+    val testInputs = checkIfStdInputExist(mockInputSectionStderr, StderrElement, "stderr")
+    testInputs match {
+      case Valid(_) => fail("Input section contained stderr. Should have failed.")
+      case Invalid(e) => e.head should be("Workflow cannot have stderr expression in input section at workflow-level.")
+    }
+  }
+
+  it should "not throw an error for non-stdout/stderr in the inputs section" in {
```

Comment:
One nice thing about the validations is that they don't ever _throw_ things, the error is wrapped up in the return type to be handled whenever you want to handle it.

```suggestion
  it should "not return an error for non-stdout/stderr in the inputs section" in {
```